src/config/ch.sbb.direktverbindungen/DvFeatureInfo/DvFeatureInfo.js

```diff
@@ -0,0 +1,271 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { unByKey } from 'ol/Observable';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { makeStyles, Divider } from '@material-ui/core';
+import MenuItem from '../../../components/Menu/MenuItem';
+import usePrevious from '../../../utils/usePrevious';
+import DvLineInfo from '../DvLineInfo';
+import DvLineTitle from '../DvLineTitle';
+import removeDuplicates, {
+  getId,
+} from '../../../utils/removeDuplicateFeatures';
+import parseDvFeatures from '../../../utils/dvParseFeatures';
+import { DV_DAY_NIGHT_REGEX, DV_KEY } from '../../../utils/constants';
+
+const useStyles = makeStyles(() => {
+  return {
+    root: {
+      '&.wkp-menu-item': {
+        marginTop: '0 !important',
+        border: 'none !important',
+      },
+      '& .wkp-menu-item-header': {
+        height: '40px !important',
+        '&.open': {
+          borderBottom: 'none !important',
+        },
+      },
+      '& .wkp-menu-item-header-toggler': {
+        marginRight: 5,
+      },
+    },
+    teaser: {
+      maxHeight: 400,
+      position: 'relative',
+      overflow: 'hidden',
+      '&::after': {
+        content: '""',
+        position: 'absolute',
+        zIndex: 10,
+        bottom: 0,
+        left: 0,
+        pointerEvents: 'none',
+        backgroundImage:
+          'linear-gradient(to bottom, rgba(255,255,255,0), rgba(255,255,255, 1) 90%)',
+        width: '100%',
+        height: '10em',
+      },
+      '& .open': {
+        '& .wkp-menu-item-header-toggler': {
+          transform: 'rotate(180deg)',
+        },
+      },
+    },
+    featureInfos: {
+      maxHeight: '100%',
+      overflow: 'auto',
+    },
+    featureInfoItem: {
+      marginLeft: 38,
+      marginRight: 16,
+    },
+    imageLine: {
+      '& img': {
+        width: '100%',
+      },
+    },
+  };
+});
+
+function DvFeatureInfo({ filterByType }) {
+  const featureInfo = useSelector((state) => state.app.featureInfo);
+  const layers = useSelector((state) => state.map.layers);
+  const [infoKey, setInfoKey] = useState();
+  const [teaser, setTeaser] = useState(true);
+  const [expandedHeight, setExpandedHeight] = useState();
+  const classes = useStyles({ teaser });
+  const teaserOnClick = useCallback(() => {
+    return teaser ? setTeaser(false) : undefined;
+  }, [teaser]);
+  const [revision, forceRender] = useState();
+
+  const getVisibleLayerKeys = useCallback(
```

Comment:
like in previous component , this callback sems overkill, memoized layersVisible instead.

Reply:
Done